src/scores/continuous/threshold_weighted_impl.py

```diff
@@ -0,0 +1,739 @@
+    a trapezoidal weight function. The weight is 1 on the interval (b, c) and 0 outside
+    the interval (a,d). The formula is based on the fourth row of Table B1 from Taggart (2022).
+
+    Args:
+        a: left endpoint of interval where weight > 0.
+        b: left endpoint of the interval where weight = 1.
+        c: right endpoint of the interval where weight = 1.
+        d: right endpoint of interval where weight > 0.
+        x: points where phi_j is to be evaluated.
+
+    Returns:
+        array of function values for the function phi_j.
+
+    Note:
+        - Requires a < b < c < d. This is tested in `_auxiliary_funcs`.
+        - phi is this case is 2x^2 to be consistent with Taggart (2022) and is used to
+            produce a twMSE. This means that a scaling factor is introduced for the
+            threshold weighted expectile and Huber Loss scores.
+    """
+    # results correspond to each case in the fourth row of Table B1, Taggart (2022).
+    result0 = 0
+    result1 = 2 * (x - a) ** 3 / (3 * (b - a))
+    result2 = 2 * x**2 - 2 * (a + b) * x + 2 * (b - a) ** 2 / 3 + 2 * a * b
+    result3 = (
+        2 * (d - x) ** 3 / (3 * (d - c))
+        + 2 * (d + c - a - b) * x
+        + 2 * ((b - a) ** 2 + 3 * a * b - (d - c) ** 2 - 3 * c * d) / 3
+    )
+    result4 = 2 * (d + c - a - b) * x + 2 * ((b - a) ** 2 + 3 * a * b - (d - c) ** 2 - 3 * c * d) / 3
+
+    result = result1.where(x >= a, result0)
+    result = result.where(x < b, result2)
+    result = result.where(x < c, result3)
+    result = result.where(x < d, result4)
+    result = result.where(~np.isnan(x), np.nan)
+
+    return result
+
+
+def _phi_j_prime_trap(
+    a: EndpointType, b: EndpointType, c: EndpointType, d: EndpointType, x: xr.DataArray
+) -> xr.DataArray:
+    """
+    Calculates the subderivative of `_phi_j_trap(a, b, c, d, x)` w.r.t. x.
+
+    Args:
+        a: left endpoint of interval where weight > 0.
+        b: left endpoint of the interval where weight = 1.
+        c: right endpoint of the interval where weight = 1.
+        d: right endpoint of interval where weight > 0.
+        x: points where phi_j is to be evaluated.
+
+    Returns:
+        The subderivative of `_phi_j_trap(a, b, c, d, x)` w.r.t. x.
+    """
+    return 4 * _g_j_trap(a, b, c, d, x)
+
+
+def tw_squared_error(
+    fcst: xr.DataArray,
+    obs: xr.DataArray,
+    interval_where_one: Tuple[EndpointType, EndpointType],
+    *,
+    interval_where_positive: Optional[Tuple[EndpointType, EndpointType]] = None,
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    weights: Optional[xr.DataArray] = None,
+) -> xr.DataArray:
+    """
+    Returns the the threshold weighted squared error.
+
+    For more flexible weighting schemes,
+    see :py:func:`scores.continuous.consistent_expectile_score`.
+
+    Two types of threshold weighting are supported: rectangular and trapezoidal.
+
```

Comment:
Please remove the blank line on line 327. This helps with the sphinx rendering of the API docs in this situation (the whens and whys of blank lined, indentation and bullet lists are very nuanced)